services/worker/src/worker/job_runners/split/duckdb_index.py

```diff
@@ -0,0 +1,263 @@
+            )
+
+        parquet_urls = [parquet_file["url"] for parquet_file in split_parquet_files]
+
+        if not parquet_urls:
+            raise ParquetResponseEmptyError("No parquet files found.")
+
+        # get the features
+        features = content_parquet_and_info["dataset_info"]["features"]
+        column_names = ",".join(list(features.keys()))
+
+        # look for string columns
+        string_columns = [
+            column
+            for column, feature in features.items()
+            if "dtype" in feature
+            and "_type" in feature
+            and feature["dtype"] == STRING_FEATURE_DTYPE
+            and feature["_type"] == VALUE_FEATURE_TYPE
+        ]
+        if not string_columns:
+            raise NoIndexableColumnsError("No string columns available to index.")
+
+    except KeyError as e:
+        raise PreviousStepFormatError(
+            f"Previous step '{config_parquet_and_info_step}' did not return the expected content.", e
+        ) from e
+
+    # configure duckdb extensions
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="httpfs"))
+    duckdb.execute(INSTALL_EXTENSION_COMMAND.format(extension="fts"))
+    duckdb.execute(LOAD_EXTENSION_COMMAND.format(extension="fts"))
+
+    # index all columns
+    if duckdb_index_file_directory is None:
+        raise CachedDirectoryNotInitializedError("Cache directory has not been initialized.")
+    db_path = Path(duckdb_index_file_directory.resolve() / DUCKDB_DEFAULT_INDEX_FILENAME)
+
+    con = duckdb.connect(str(db_path.resolve()))
+    logging.debug(CREATE_SEQUENCE_COMMAND)
+    con.sql(CREATE_SEQUENCE_COMMAND)
+
+    create_command_sql = f"{CREATE_TABLE_COMMAND.format(columns=column_names)} read_parquet({parquet_urls});"
+    logging.debug(create_command_sql)
+    con.sql(create_command_sql)
+
+    # TODO: by default, 'porter' stemmer is being used, use a specific one by dataset language in the future
+    # see https://duckdb.org/docs/extensions/full_text_search.html for more details about 'stemmer' parameter
+    logging.debug(CREATE_INDEX_COMMAND)
+    con.sql(CREATE_INDEX_COMMAND)
+    con.close()
+
+    # create the target revision if it does not exist yet (clone from initial commit to avoid cloning all repo's files)
+    hf_api = HfApi(endpoint=hf_endpoint, token=hf_token)
+    committer_hf_api = HfApi(endpoint=hf_endpoint, token=committer_hf_token)
+    index_file_location = f"{config}/{split}/{DUCKDB_DEFAULT_INDEX_FILENAME}"
+    try:
+        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
+    except RepositoryNotFoundError as err:
+        raise DatasetNotFoundError("The dataset does not exist on the Hub.") from err
+
+    create_branch(
+        dataset=dataset, target_revision=target_revision, refs=refs, hf_api=hf_api, committer_hf_api=committer_hf_api
+    )
+
+    try:
+        sleeps = [1, 1, 1, 10, 10, 100, 100, 100, 300]
```

Comment:
please use the same sleeps as config-parquet-and-info (I think we could move this sleeps definition to utils.py)